tests/core/fsck-vfat/task.yaml

```diff
@@ -0,0 +1,68 @@
+summary: the boot base provides essential fsck programs
+details: |
+  Snapd uses vfat on certain essential boot partitions, due to external
+  requirements imposed by the bootloader architecture. This test verifies
+  that fsck.vfat is shipped in the image, and that it can correctly clean
+  the dirty bit that is artificially set by this test.
+
+  A separate test examines how fsck is automatically invoked during the boot
+  process. This is not verified here.
+prepare: |
+  tests.cleanup prepare
+execute: |
+  echo "Essential fsck programs are in the boot base"
```

Comment:
we should also add tests for `mkfs.vfat`, etc. too I think we had a bug where core20 didn't actually have `mkfs.vfat` that was only discovered when enabling the pi and would have been nice to catch earlier in the dev process (at that point we were running spread tests for amd64)

Reply:
I'll do this in a follow-up